### PR TITLE
[Dataset] KarateClubDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Description
KarateClubDataset inherits from DGLDataset. This dataset is transformed from networkx, which is hard-coded. So no need to download and save to local disk.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/__init__.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from .ppi import PPIDataset, LegacyPPIDataset
 from .tu import TUDataset, LegacyTUDataset
 from .gnn_benckmark import AmazonCoBuy, CoraFull, Coauthor
-from .karate import KarateClub
+from .karate import KarateClubDataset
```
Comment (hetong007) [inline python/dgl/data/__init__.py:14]: Doesn't it break any example/tutorial?

Comment (HuXiangkun) [reply python/dgl/data/__init__.py:14]: I searched no examples or tutorials using KarateClub class.

Comment (hetong007) [reply python/dgl/data/__init__.py:14]: In case any third-party's using this dataset, I suggest we keep the old named with a warning of the deprecation.

**Comment by VoVAllen** [comment]
Same opinion as @hetong007 . Other LGTM

**Pull request merged**